Assets/AY_Scripts/ParticlesService.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ParticlesService : MonoSingletonGeneric<ParticlesService>
+{
+    [SerializeField] private ParticleSystem tankExplosion;
```

Comment:
List<ParticleSystem>

Reply:
Instantiate(List[(int)ExplosionEnum.Tank) - This will only work if the order of elements in your list matches the order in the enum